src/main/java/uk/gov/hmcts/reform/jobscheduler/jobs/HttpCallJob.java

```diff
@@ -4,30 +4,49 @@
 import org.quartz.JobExecutionContext;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
-import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
 import uk.gov.hmcts.reform.jobscheduler.model.HttpAction;
-import uk.gov.hmcts.reform.jobscheduler.services.jobs.ActionSerializer;
 
 @Component
 public class HttpCallJob implements Job {
 
     public static final String PARAMS_KEY = "params";
     private static final Logger logger = LoggerFactory.getLogger(HttpCallJob.class);
 
-    private final ActionSerializer serializer;
+    private final RestTemplate restTemplate;
+    private final ActionExtractor actionExtractor;
 
-    @Autowired
-    public HttpCallJob(ActionSerializer serializer) {
-        this.serializer = serializer;
+    public HttpCallJob(RestTemplate restTemplate, ActionExtractor actionExtractor) {
+        this.restTemplate = restTemplate;
+        this.actionExtractor = actionExtractor;
     }
 
     @Override
     public void execute(JobExecutionContext context) {
-        // todo: send a http request based on data from job details
-        String json = context.getJobDetail().getJobDataMap().getString(PARAMS_KEY);
-        HttpAction actionToPerform = serializer.deserialize(json);
+        HttpAction action = actionExtractor.extract(context);
 
-        logger.info("Executing job: " + actionToPerform.url);
+        logger.info("Sending request to " + action.url);
+
+        ResponseEntity<String> response =
+            restTemplate
+                .exchange(
+                    action.url,
+                    action.method,
+                    toHttpEntity(action),
+                    String.class
+                );
+
+        logger.info("Response code: " + response.getStatusCodeValue());
+    }
+
+    private static HttpEntity toHttpEntity(HttpAction action) {
```

Comment:
I was wondering if this method could be moved into `HttpAction` as that class has everything required for preparing `HttpEntity` object.

Reply:
It could, but it would couple it with `org.springframework.http.HttpEntity`.
I prefer to keep it here.